templates/netlify/remix.init/index.js

```diff
@@ -0,0 +1,102 @@
+const inquirer = require("inquirer");
+const fs = require("fs/promises");
+const { join } = require("path");
+const PackageJson = require("@npmcli/package-json");
+
+const filesToCopy = [
+  ["README.md"],
+  ["netlify.toml"],
+  ["server.js"],
+  ["remix.config.js"],
+  ["vscode.json", join(".vscode", "settings.json")],
+  ["globals.d.ts"],
+];
+
+const filesToModify = ["app/entry.server.tsx", "app/root.tsx"];
+
+async function modifyFilesForEdge(files, rootDirectory) {
+  const filePaths = files.map((file) => join(rootDirectory, file));
+  const contents = await Promise.all(
+    filePaths.map((path) => fs.readFile(path, "utf8"))
+  );
+
+  await Promise.all(
+    contents.map((content, index) => {
+      const newContent = content.replaceAll(
+        "@remix-run/node",
+        "@remix-run/deno"
+      );
+      return fs.writeFile(filePaths[index], newContent);
+    })
+  );
+}
+
+async function copyEdgeTemplateFiles(files, rootDirectory) {
+  for (const [file, target] of files) {
+    await fs.copyFile(
+      join(rootDirectory, "remix.init", file),
+      join(rootDirectory, target || file)
+    );
+  }
+}
+
+async function updatePackageJsonForEdge(directory) {
+  const packageJson = await PackageJson.load(directory);
+  const {
+    dependencies: {
+      "@remix-run/netlify": _netlify,
+      "@remix-run/node": _node,
+      ...dependencies
+    },
+    scripts,
+    ...restOfPackageJson
+  } = packageJson.content;
+
+  packageJson.update({
+    // dev script is the same as the start script for Netlify Edge, "cross-env NODE_ENV=production netlify dev"
+    scripts: { ...scripts, dev: scripts["start"] },
+    ...restOfPackageJson,
+    dependencies: { ...dependencies, "@remix-run/netlify-edge": "*" },
```

Comment:
It's also used in `app/entry.server.tsx` & `app/root.tsx` as we replace `@remix-run/node` with `@remix-run/deno` in `modifyFilesForEdge`

Reply:
Oh yeah, good catch. I think it builds and deploys fine because it's a dependency of `@remix-run/netlify-edge`.

Looking at `@remix-run/deno`, that type is imported from `@remix-run/server-runtime` and then exported from https://github.com/remix-run/remix/blob/main/packages/remix-deno/index.ts#L28-L70

I'm wondering if it make more sense to follow that pattern so those types are available from https://github.com/nickytonline/remix/blob/48d76d57b4100348eed22f0656a333206ba46549/packages/remix-netlify-edge/index.ts and then `@remix-run/deno` can stay internal to `@remix-run/netlify-edge`.

Reply:
This case makes it indeed clear that we have something strange going on

@pcattori already created a proposal for this: https://github.com/remix-run/remix/discussions/4634
But as long as we don't have the separate package, we should keep everything the way it's setup now

Reply:
If my comment above (https://github.com/remix-run/remix/pull/3104#discussion_r1034094380) is correct, with Netlify Edge, it's the first time the runtime is an implementation detail/something internal

Reply:
Not sure when that proposal will land/be accepted, so in the meantime, should I export from `@remix-run/server-runtime` instead?

Reply:
No, using `@remix-run/server-runtime` is seen as an anti-pattern. We should import from `@remix-run/cloudflare`, `@remix-run/deno` or `@remix-run/node`

Reply:
OK. I'll explicitly add `@remix-run/deno` for now while that proposal is being discussed.